File: crates/backend-domain/src/pinned_message.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

use crate::{ChannelId, MessageId, PinnedMessageId, ServerId, UserId};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
pub struct PinnedMessage {
    pub id: PinnedMessageId,
    pub server_id: ServerId,
    pub channel_id: ChannelId,
    pub message_id: MessageId,
    pub pinned_by_user_id: UserId,
    pub content: String,
    pub author_user_id: UserId,
}

File: apps/flutter_client/lib/shared/models/pinned_message.dart
import "package:polyphony_flutter_client/shared/models/entity_ids.dart";

class PinnedMessage {
  const PinnedMessage({
    required this.id,
    required this.serverId,
    required this.channelId,
    required this.messageId,
    required this.pinnedByUserId,
    required this.content,
    required this.authorUserId,
  });

  final String id;
  final ServerId serverId;
  final ChannelId channelId;
  final MessageId messageId;
  final UserId pinnedByUserId;
  final String content;
  final UserId authorUserId;
}
